R/knitr-engine.R

```diff
@@ -209,13 +209,26 @@ eng_python_initialize <- function(options, context, envir) {
 }
 
 eng_python_matplotlib_show <- function(plt, options) {
+  
+  # work within the knitr 'base dir', to ensure that figures are
+  # generated in the same location where R figures might normally
+  # be generated
+  #
+  # https://github.com/rstudio/reticulate/issues/645
+  dir <- knitr::opts_knit$get("output.dir")
+  if (is.character(dir)) {
+    dir.create(dir, recursive = TRUE)
```

Comment:
this creates a warning I think because `output.dir` will always exist. 

I don't think it is necessary to test it. At least you could `&& !file_test("-d", dir)`